src/fallback.rs

```diff
@@ -75,8 +75,16 @@ pub fn fish_wcwidth(c: char) -> isize {
     let width = WC_LOOKUP_TABLE.classify(c);
     match width {
         WcWidth::NonCharacter | WcWidth::NonPrint | WcWidth::Combining | WcWidth::Unassigned => {
-            // Fall back to system wcwidth in this case.
-            wcwidth(c)
+            #[cfg(not(target_os = "cygwin"))]
+            {
+                // Fall back to system wcwidth in this case.
+                wcwidth(c)
+            }
+            #[cfg(target_os = "cygwin")]
+            {
+                // No system wcwidth for UTF-32 on cygwin.
+                0
```

Comment:
should this be Ambiguous instead of 0?

Reply:
They are 0 in cygwin. You can see in the following code that it returns 0 for non-spacing characters.

https://github.com/cygwin/cygwin/blob/main/newlib/libc/string/wcwidth.c

We don't use the cygwin `wcwidth` function because it's UTF-16.